data/modules/PolicePatrol/PolicePatrol.lua

```diff
@@ -140,37 +141,58 @@ local onEnterSystem = function (player)
 		label = system.faction.policeName
 	}
 
-	for i = 1, n do
-		ship = ShipBuilder.MakeShipNear(player, template, threat, 50, 100)
-		assert(ship)
+	-- The scene is set for a police patrol, just wait!
+	Timer:CallAt(Game.time + Engine.rand:Integer(5, 10), function ()
+		if not Game.system then return end -- Shut up when the player is already in hyperspace
 
-		table.insert(patrol, ship)
-	end
+		for i = 1, n do
+			ship = ShipBuilder.MakeShipNear(player, template, threat, 50, 100) -- "Ship detected nearby"
+			assert(ship)
 
-	if Engine.rand:Number(1) < system.lawlessness then
-		-- You are lucky. They are busy, eating donuts ;-)
-		Comms.ImportantMessage(string.interp(l["RESPECT_THE_LAW_" .. Engine.rand:Integer(1, getNumberOfFlavours("RESPECT_THE_LAW"))], { system = system.name }), ship.label)
-	else
-		if fine > maxFineTolerated then
-			Comms.ImportantMessage(string.interp(l["OUTLAW_DETECTED_" .. Engine.rand:Integer(1, getNumberOfFlavours("OUTLAW_DETECTED"))], { ship_label = player.label }), ship.label)
-			showMercy = false
-			attackShip(player)
-		else
-			Comms.ImportantMessage(l["INITIATE_CARGO_SCAN_" .. Engine.rand:Integer(1, getNumberOfFlavours("INITIATE_CARGO_SCAN"))], ship.label)
-			Timer:CallAt(Game.time + Engine.rand:Integer(3, 9), function ()
-				if not Game.system then return end -- Shut up when the player is already in hyperspace
-
-				local manifest = player:GetComponent('CargoManager').commodities
-				if hasIllegalGoods(manifest) then
-					Comms.ImportantMessage(l.ILLEGAL_GOODS_DETECTED, ship.label)
+			table.insert(patrol, ship)
+		end
+
+		Timer:CallAt(Game.time + Engine.rand:Integer(5, 10), function () -- Oh crap, it's the police!
+			if not Game.system then return end -- Shut up if the player has jumped away
+
+			if Engine.rand:Number(1) < system.lawlessness then
+				-- You are lucky. They are busy, eating donuts ;-)
+				Comms.ImportantMessage(string.interp(l["RESPECT_THE_LAW_" .. Engine.rand:Integer(1, getNumberOfFlavours("RESPECT_THE_LAW"))], { system = system.name }), ship.label)
+			else
+				if fine > maxFineTolerated then
+					Comms.ImportantMessage(string.interp(l["OUTLAW_DETECTED_" .. Engine.rand:Integer(1, getNumberOfFlavours("OUTLAW_DETECTED"))], { ship_label = player.label }), ship.label)
+					showMercy = false
 					attackShip(player)
-					Comms.ImportantMessage(l["POLICE_TAUNT_" .. Engine.rand:Integer(1, getNumberOfFlavours("POLICE_TAUNT"))], ship.label)
 				else
-					Comms.ImportantMessage(l.NOTHING_DETECTED, ship.label)
+					Comms.ImportantMessage(l["INITIATE_CARGO_SCAN_" .. Engine.rand:Integer(1, getNumberOfFlavours("INITIATE_CARGO_SCAN"))], ship.label)
+					Timer:CallAt(Game.time + Engine.rand:Integer(3, 9), function ()
+						if not Game.system then return end -- Shut up when the player is already in hyperspace
```

Comment:
Similarly, this should sanity-check that the player is in the same system. Failure to do so may cause invalid-object access errors on `ship`.

Reply:
fixed and fixed.